### PR TITLE
Rep-8 Create API Response Tests

### PR DESCRIPTION
**Pull request by kcover**
#### What does this PR do?
Creates tests to confirm that the API sends the proper responses. It also fixes an issue I found while creating the tests where the hostname and port of an object were returned as null.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@paouelle @peterhuffer @clockard 

#### How should this be tested? (List steps with links to updated documentation)
Start an instance of DDF and run the tests.

#### What are the relevant tickets?
#8 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [x] Update / Add Integration Tests

### CODE REVIEW

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  public void createReplicationSiteWithIntegerName() {
+    asAdmin()
+        .body(
+            String.format(
+                "{\"query\":\"mutation{ createReplicationSite(name: 25, address: { url: \\\"%s\\\"}){ id name address{ host{ hostname port } url }}}\"}",
+                URL))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
```
Comment (paouelle) [inline distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: ❗️ I would say returning 200 OK when a request is invalid is a problem. This is not something a client of an API would expect.

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+  @Test
+  public void createReplicationSite() throws MalformedURLException {
+    String name = "create";
+    asAdmin()
+        .body(makeCreateSiteQuery(name, URL))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
```
Comment (paouelle) [inline distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: ❓ Should we verify the content-type of the responses?

File: distributions/test/pom.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (paouelle) [inline distributions/test/pom.xml]: ❓ Should we follow the DDF convention and define the project hierarchy for tests as: distributions/test/itests?

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java (hunk 1 above)
Comment (peterhuffer) [reply distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: When type validation fails, GraphQL returns a 200 but the "errors" field in the JSON response is populated.

File: distributions/test/pom.xml (hunk 3 above)
Comment (kcover) [reply distributions/test/pom.xml]: I tried to address this one already. Is there something that I missed?

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java (hunk 1 above)
Comment (kcover) [reply distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: I would think that a 400 would make more sense, but I think that is a problem with the endpoint, not with the tests.

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java (hunk 2 above)
Comment (kcover) [reply distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: Is it really possible to return anything other than json? And, would that be a test of it's own, or an assertion we make in each test?

Comment (paouelle) [reply distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: Is there a content-type header in the response? There should be right?

File: distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java (hunk 1 above)
Comment (paouelle) [reply distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java]: it depends. I agree that 400 makes more sense to me as well. If that is what we say our API should return then we should validate that in our test and let the test fail until we fix it. If it turns out that 200 is valid in this case as we specify an error code inside the response message than the test case is ok. It all depends on what we document our API/protocol to be.

File: distributions/test/pom.xml (hunk 3 above)
Comment (paouelle) [reply distributions/test/pom.xml]: Should there be another level underneath `test` named `itests` with its own pom file and the test class defined there?

Comment (kcover) [reply distributions/test/pom.xml]: There is. The module is declared just below this comment 🙂

Comment (paouelle) [reply distributions/test/pom.xml]: Oops. I guess I just carried over this comment from the old repo 😆
